caproto/_log.py

```diff
@@ -143,10 +155,128 @@ def color_logs(color):
+            if fnmatch.fnmatch(record.name, i) and record.levelno >= self.levelno:
+                return True
+        return False
+
+
+def validate_level(level)-> int:
+    '''
+    Return a int for level comparison
+
+    '''
+    if isinstance(level, int):
+        levelno = level
+    elif isinstance(level, str):
+        levelno = logging.getLevelName(level)
+
+    if isinstance(levelno, int):
+        return levelno
+    else:
+        raise CaprotoValueError("The target addresses should given as a list of strings "
+                "like 'XX.XX.XX.XX:YYYY' "
+                "or tuples like ('XX.XX.XX.XX', YYYY).")
+
+class PVFilter(logging.Filter):
+    '''
+    Pass through only messages relevant to one or more PV names and
+    pv-unrelated messages.
+
+    Parameters
+    ----------
+    names : string or list of string
+        PV name or PV name list which will be filtered in.
+
+    Returns
+    -------
+    Bool: True or False
+        True if message is not PV related which doesn't have 'pv' as key in extra
+        True if 'pv' as key exists and pv name exists in Filter list.
+        False if message is PV related but pv name isn't in Filter list.
+    '''
+    def __init__(self, names, level='NOTSET', exclusive=False):
+        self.names = names
+        self.levelno = validate_level(level)
+        self.exclusive = exclusive
+
+    def filter(self, record):
+        if record.levelno < self.levelno:
+            return False
+        elif hasattr(record, 'pv'):
+            for i in self.names:
+                if fnmatch.fnmatch(record.pv, i):
+                    return True
+            return False
+        else:
+            return not self.exclusive
+
+
+class AddressFilter(logging.Filter):
+
+    def __init__(self, addresses_list, level='NOTSET', exclusive=False):
+        self.addresses_list = []
+        self.hosts_list = []
+        for address in addresses_list:
+            if isinstance(address, str):
+                if ':' in address:
+                    host, port_as_str = address.split(':')
+                    self.addresses_list.append((host, int(port_as_str)))
+                else:
+                    self.hosts_list.append(address)
+            elif isinstance(address, tuple):
+                if len(address) == 2:
+                    self.addresses_list.append(address)
+                else:
+                    raise CaprotoValueError("The target addresses should given as a list of strings "
+                            "like 'XX.XX.XX.XX:YYYY' "
+                            "or tuples like ('XX.XX.XX.XX', YYYY).")
+            else:
+                raise CaprotoValueError("The target addresses should given as a list of strings "
+                        "like 'XX.XX.XX.XX:YYYY' "
+                        "or tuples like ('XX.XX.XX.XX', YYYY).")
+
+        self.levelno = validate_level(level)
+        self.exclusive = exclusive
+
+    def filter(self, record):
```

Comment:
Same here. Please use the agreed-upon formulation of the filter, with the comments, changing only the part

```python
for i in self.names:
    if  fnmatch.fnmatch(record.pv, i):
        # The record satisfies the condition.
        return True
```

to use the filtering logic here.